sql/moz-fx-data-shared-prod/telemetry_derived/firefox_android_clients_v1/query.sql

```diff
@@ -0,0 +1,482 @@
+-- Query first observations for Firefox Android Clients.
+-- Proposal:
+-- https://docs.google.com/document/d/12bj4DhCybelqHVgOVq8KJlzgtbbUw3f68palNrv-gaM/edit#
+CREATE TEMP FUNCTION get_first_not_null_value(
+  existing_value STRING,
+  existing_date DATE,
+  first_session_value STRING,
+  first_session_date DATE,
+  metrics_value STRING,
+  metrics_date DATE
+) AS (
+  -- Return first not null value, or null.
+  CASE
+  WHEN
+    existing_value IS NULL
+  THEN
+    COALESCE(first_session_value, metrics_value)
+  WHEN
+    existing_value IS NOT NULL
+    AND first_session_value IS NOT NULL
+    AND first_session_date < existing_date
+  THEN
+    first_session_value
+  WHEN
+    existing_value IS NOT NULL
+    AND first_session_value IS NULL
+    AND metrics_value IS NOT NULL
+    AND metrics_date < existing_date
+  THEN
+    metrics_value
+  ELSE
+    existing_value
+  END
+);
+
+CREATE TEMP FUNCTION get_first_not_null_date(
+  existing_value STRING,
+  existing_date DATE,
+  first_session_value STRING,
+  first_session_date DATE,
+  metrics_value STRING,
+  metrics_date DATE
+) AS (
+  -- Return the date of the first source ping where adjust_network is not null.
+  CASE
+  WHEN
+    existing_value IS NULL
+    AND first_session_value IS NOT NULL
+  THEN
+    first_session_date
+  WHEN
+    existing_value IS NULL
+    AND first_session_value IS NULL
+    AND metrics_value IS NOT NULL
+  THEN
+    metrics_date
+  WHEN
+    existing_value IS NOT NULL
+    AND first_session_value IS NOT NULL
+    AND first_session_date < existing_date
+  THEN
+    first_session_date
+  WHEN
+    existing_value IS NOT NULL
+    AND first_session_value IS NULL
+    AND metrics_value IS NOT NULL
+    AND metrics_date < existing_date
+  THEN
+    metrics_date
+  ELSE
+    existing_date
+  END
+);
+
+-- Return the source ping that reported the first not null value
+CREATE TEMP FUNCTION get_source_ping(
+  existing_source STRING,
+  existing_date DATE,
+  first_session_value STRING,
+  first_session_date DATE,
+  metrics_value STRING,
+  metrics_date DATE
+) AS (
+  CASE
+  WHEN
+    existing_source IS NULL
+    AND first_session_value IS NOT NULL
+  THEN
+    "first_session"
+  WHEN
+    existing_source IS NULL
+    AND first_session_value IS NULL
+    AND metrics_value IS NOT NULL
+  THEN
+    "metrics"
+  WHEN
+    existing_source IS NOT NULL
+    AND first_session_value IS NOT NULL
+    AND first_session_date < existing_date
+  THEN
+    "first_session"
+  WHEN
+    existing_source IS NOT NULL
+    AND first_session_value IS NULL
+    AND metrics_value IS NOT NULL
+    AND metrics_date < existing_date
+  THEN
+    "metrics"
+  ELSE
+    existing_source
+  END
+);
+
+WITH first_seen AS (
+  SELECT
+    client_id,
+    first_seen_date
+  FROM
+    `mozdata.org_mozilla_firefox.baseline_clients_first_seen`
```

Comment:
Looks like it's there, runs with [copy_deduplicate](https://workflow.telemetry.mozilla.org/graph?dag_id=copy_deduplicate)

Reply:
Yes, but [with a different task_id](https://github.com/mozilla/telemetry-airflow/blob/main/dags/copy_deduplicate.py#L305-L326). The view for `baseline_clients_first_seen` is actually over the `baseline_clients_daily` table, so that will have to be the dependency.